src/CreateSignalFactory.js

```diff
@@ -150,45 +139,45 @@ module.exports = function (controller, model, services, compute, modules) {
               runBranch(branch, index + 1)
             } else {
               var promises = currentBranch.map(function (action) {
-                controller.emit('actionStart', {action: action, signal: signal})
+                controller.emit('actionStart', {action: action, signal: signal, options: options, payload: payload})
                 var actionFunc = actions[action.actionIndex]
                 var inputArg = actionFunc.defaultInput ? utils.merge({}, actionFunc.defaultInput, signalArgs) : signalArgs
-                var actionArgs = createActionArgs.async(action, inputArg, model, compute, services, Object.keys(modules))
+                var next = createNext.async(actionFunc, signal.name)
+                var contextProviders = [
+                  inputProvider(inputArg),
```

Comment:
It looks like context provider factories, but not context providers.
Maybe better call it like `createInputProvider`?